File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/util/FileUtils.java
package org.javamaster.invocationlab.admin.util;

import lombok.Cleanup;
import lombok.SneakyThrows;
import lombok.extern.slf4j.Slf4j;

import java.io.InputStream;
import java.net.URL;
import java.net.URLConnection;
import java.nio.charset.StandardCharsets;

/**
 * @author yudong
 */
@Slf4j
public class FileUtils {

    @SneakyThrows
    public static String readStringFromUrl(URL url) {
        if (url == null) {
            return null;
        }
        URLConnection urlConnection = url.openConnection();
        urlConnection.setUseCaches(false);
        @Cleanup
        InputStream in = urlConnection.getInputStream();
        int len = in.available();
        byte[] data = new byte[len];
        int read = in.read(data, 0, len);
        log.info("read url:{} length:{}", url, read);
        return new String(data, StandardCharsets.UTF_8);
    }

}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/util/FileUtils.kt
package org.javamaster.invocationlab.admin.util

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import java.net.URL
import java.nio.charset.StandardCharsets

/**
 * @author yudong
 */

object FileUtils {
    private val log: Logger = LoggerFactory.getLogger(this.javaClass)

    @JvmStatic
    fun readStringFromUrl(url: URL): String {
        val urlConnection = url.openConnection()
        urlConnection.useCaches = false
        urlConnection.getInputStream().use {
            val len = it.available()
            val data = ByteArray(len)
            val read = it.read(data, 0, len)
            log.info("read url:{} length:{}", url, read)
            return String(data, StandardCharsets.UTF_8)
        }
    }
}
